satpy/readers/fci_l1c_fdhsi.py

```diff
@@ -220,55 +268,75 @@ def _get_dataset_measurand(self, key, info=None):
 
         return res
 
-    def _get_dataset_quality(self, key, info=None):
-        """Load quality for channel.
-
-        Load a quality field for an FCI channel.  This is a bit involved in
-        case of FCI because each channel group (data/<channel>/measured) has
-        its own data variable 'pixel_quality', referred to in ancillary
-        variables (see also Satpy issue 1171), so some special treatment in
-        necessary.
-        """
-        # FIXME: replace by .removesuffix after we drop support for Python < 3.9
-        if key['name'].endswith("_pixel_quality"):
-            chan_lab = key['name'][:-len("_pixel_quality")]
-        else:
-            raise ValueError("Quality label must end with pixel_quality, got "
-                             f"{key['name']:s}")
-        grp_path = self.get_channel_measured_group_path(chan_lab)
+    def _get_dataset_quality(self, dsname):
+        """Load a quality field for an FCI channel."""
+        grp_path = self.get_channel_measured_group_path(_get_channel_name_from_dsname(dsname))
         dv_path = grp_path + "/pixel_quality"
         data = self[dv_path]
         return data
 
-    def get_channel_measured_group_path(self, channel):
+    def _get_dataset_index_map(self, dsname):
+        """Load the index map for an FCI channel."""
+        grp_path = self.get_channel_measured_group_path(_get_channel_name_from_dsname(dsname))
+        dv_path = grp_path + "/index_map"
+        data = self[dv_path]
+
+        data = data.where(data != data.attrs.get('_FillValue', 65535))
+        return data
+
+    @staticmethod
+    def _getitem(block, lut):
+        return lut[block.astype('uint16')]
+
+    def _get_dataset_aux_data(self, dsname):
+        """Get the auxiliary data arrays using the index map."""
+        # get index map
+        index_map = self._get_dataset_index_map(_get_channel_name_from_dsname(dsname))
+        # index map indexing starts from 1
+        index_map -= 1
```

Comment:
I think this belongs to the `_get_dataset_index_map` method.

Reply:
I would actually like to keep it here: this way, a user that is only interested in reading the index map for his own use, gets the values as they're stored in the file and described in the docs. 
The `-= 1` is only relevant when using the index map in combination with the lut (so in `_get_dataset_aux_data`), as the (python) indexing needs to start from 0.